test/location.test.ts

```diff
@@ -33,4 +39,81 @@ describe('browser/location', () => {
       })
     );
   });
+
+  it('blocking transitions should work', async () => {
+    await doUpdateLocation({
+      pathname: '/',
+      search: {}
+    }).run();
+
+    let unblock = requestConfirmationToUpdateLocation();
+    await doUpdateLocation({
+      pathname: '/foo',
+      search: { bar: 'baz' }
+    }).run();
+    expect(await location.run().run()).toEqual(
+      right({
+        pathname: '/',
+        search: {}
+      })
+    );
+    await doResolvePendingUpdateLocation(false).run();
+    expect(await location.run().run()).toEqual(
+      right({
+        pathname: '/',
+        search: {}
+      })
+    );
+    unblock();
+
+    unblock = requestConfirmationToUpdateLocation();
+    await doUpdateLocation({
+      pathname: '/foo',
+      search: { bar: 'baz' }
+    }).run();
+    expect(await location.run().run()).toEqual(
+      right({
+        pathname: '/',
+        search: {}
+      })
+    );
+    await doResolvePendingUpdateLocation(true).run();
+    expect(await location.run().run()).toEqual(
+      right({
+        pathname: '/foo',
+        search: { bar: 'baz' }
+      })
+    );
+    unblock();
+  });
+
+  it('should not push to history when running command with the same input', async () => {
+    await doUpdateLocation({
+      pathname: '/foo',
+      search: { foo: 'bar' }
+    }).run();
+    const fn = jest.fn();
+    observeStrict(location, undefined).subscribe(fn);
+    await new Promise(r => setTimeout(r));
+    expect(fn.mock.calls.length).toBe(2);
```

Comment:
does it make sense to add a test for `should push to history when running command with different input`?

Reply:
added